cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicSystemHttpServiceContext.java

```diff
@@ -137,4 +161,22 @@ public void run(TxRunnable runnable) throws TransactionException {
       throw new IllegalArgumentException(String.format("Namespace '%s' does not exist", namespace), nfe);
     }
   }
+
+  @Override
+  public byte[] runTask(RunnableTaskRequest runnableTaskRequest) throws IOException {
+    HttpRequest.Builder requestBuilder = remoteTaskClient.requestBuilder(HttpMethod.POST, TASK_WORKER_URL);
```

Comment:
Why it is implemented in here? I would image there are multiple places in the core system that need to run tasks. Shouldn't we have one implementation of submitting task and shared across the code base?

Reply:
Yes, will move to a common place.

Reply:
Done.